internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -34,11 +34,17 @@ var (
 // Server is a fake S2A Server for testing.
 type Server struct {
 	s2av2pb.UnimplementedS2AServiceServer
+	validToken string
+	isClientSide bool
+	// TODO(rmehta19): Decide whether to also store validationResult (bool).
```

Comment:
For my own understanding: does "validationResult" mean "yes at least one auth mech was successfully validated vs. no none of the auth mechs were successfully validated"?

Reply:
I meant for validationResult to indicate at least one auth mechanism was successfully validated. If I added this field, it would enforce that SessionReq with ```GetTlsConfigurationReq``` needs to be sent on the stream before anything else. This would mainly just affect the unit tests I have for the client libraries. Before I send a ```ValidatePeerCertificateChainReq```, I would need to send a SessionReq with a ```GetTlsConfigurationReq```

Reply:
Understood, this plan SGTM.